Fixed #28344 -- Allowed customizing queryset in Model.refresh_from_db()/arefresh_from_db(). (#17592)
django/django: docs/ref/models/instances.txt

    >>> del obj.field
    >>> obj.field  # Loads the field from the database

.. method:: Model.refresh_from_db(using=None, fields=None, from_queryset=None)

> .. method:: Model.arefresh_from_db(using=None, fields=None, from_queryset=None)

Reviewer: `versionchanged` annotation is missing, e.g.
```
.. versionchanged:: 5.1

    The `from_queryset` argument was added.
```

Reviewer: Please add `from_queryset` to the `arefresh_from_db()`.

                    fields = fields.union(deferred_fields)
            super().refresh_from_db(using, fields, **kwargs)

> The ``from_queryset`` parameter allows to use a different queryset than the one

Reviewer: ```suggestion
The ``from_queryset`` parameter allows using a different queryset than the one
```

created from ``_base_manager``. It gives you more control over how the model is
reloaded. When your model uses soft deletion you can make ``refresh_from_db``
take that into account::

    obj.refresh_from_db(from_queryset=Model.active_objects.all())

You can cache related models that otherwise would be cleared from the reloaded
instance::

>     obj.refresh_from_db(from_queryset=Model.objects.select_related('related_field')

Reviewer: Snippets should be blacked:
```suggestion
    obj.refresh_from_db(from_queryset=Model.objects.select_related("related_field")
```

You can lock the row until the end of transaction before reloading model's values::

>     obj.refresh_from_db(from_queryset=Model.objects.select_for_()

Reviewer 2: This last example looks accidentally truncated, I _assume_ it's something like:
```suggestion
    obj.refresh_from_db(from_queryset=Model.objects.select_for_update())
```
give or take arguments for `select_for_update` ...?

> 

Reviewer: Remove unnecessary blank line.
```suggestion
```

                    fields = fields.union(deferred_fields)
            super().refresh_from_db(using, fields, **kwargs)

The ``from_queryset`` parameter allows using a different queryset than the one
created from ``_base_manager``. It gives you more control over how the model is
reloaded. When your model uses soft deletion you can make ``refresh_from_db``
take that into account::

    obj.refresh_from_db(from_queryset=Model.active_objects.all())

You can cache related models that otherwise would be cleared from the reloaded
instance::

    obj.refresh_from_db(from_queryset=Model.objects.select_related("related_field")

You can lock the row until the end of transaction before reloading model's values::

>     obj.refresh_from_db(from_queryset=Model.objects.select_for_update()

Reviewer: Thanks :+1: Fixed. I'm polishing this patch :gem:

>     obj.refresh_from_db(from_queryset=Model.objects.select_for_update()

Reviewer: Edits pushed.
